eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -265,12 +291,7 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
               doPublish(c.revokedCommitPublished)
             case None =>
               // in all other cases we need to be ready for any type of closing
-              watchFundingTx(data.metaCommitments.latest.commitment, closing.spendingTxs.map(_.txid).toSet)
-              if (closing.metaCommitments.commitments.size > 1) {
-                // If we have more than one commitment, some of them must be unconfirmed: we watch all of them to be
-                // able to prune obsolete ones.
-                closing.metaCommitments.commitments.foreach(c => blockchain ! WatchFundingConfirmed(self, c.fundingTxId, nodeParams.channelConf.minDepthBlocks))
-              }
+              watchFundingSpent(data.metaCommitments.latest.commitment, closing.spendingTxs.map(_.txid).toSet)
```

Comment:
I think we still have an issue with the following scenario, let me know what you think.

- the channel's initial funding transaction is confirmed
- a splice has been initiated, and its transaction is not confirmed
- we force-close because of an error, publishing our local commit that spends the splice transaction
- when restarting, the current code sets the following watches:
  - `WatchFundingConfirmed` on the splice transaction (in the common section above)
  - `WatchFundingSpent` on the splice transaction
  - Note that we don't set `WatchFundingSpent` on the initial funding transaction
- our peer broadcasts their commit that spends the initial funding transaction, and it confirms
- we simply don't detect it and never spend our outputs in that remote commit

Reply:
Fixed in 825312b02.